packages/compass-schema/src/stores/store.ts

```diff
@@ -181,6 +184,21 @@ export function activateSchemaPlugin(
       );
     },
 
+    _trackSchemaShared(this: SchemaStore, hasSchema: boolean) {
+      const { schema } = this.state;
+      // Use a function here to a) ensure that the calculations here
+      // are only made when telemetry is enabled and b) that errors from
+      // those calculations are caught and logged rather than displayed to
+      // users as errors from the core schema sharing logic.
+      const trackEvent = () => ({
+        has_schema: hasSchema,
+        schema_width: schema?.fields?.length ?? 0,
+        schema_depth: schema ? calculateSchemaDepth(schema) : 0,
+        geo_data: schema ? schemaContainsGeoData(schema) : false,
```

Comment:
We do these same schema telemetry metadata calculations in the `_trackSchemaAnalyzed` function currently. How do you feel about using a shared function for them?

Reply:
might be worth noting - we're planning to get rid of the menu action altogether in oncoming schema-export project, so this is most likely a temporary tracking event, just to see how many people are actually using the action we're about to remove.

Reply:
@kraenhansen please confirm with Julia, but I think we don't even need any metadata

Reply:
Clarification from Julia:

> It would also be great if we include the same properties that we include in the "Schema Analyzed" event. 
> For the naming–I think it will be best to call it "Schema Exported."

Reply:
> we're planning to get rid of the menu action altogether

I believe that speaks even stronger in favour of not introducing a shared function 👍